storage/innobase/fil/fil0fil.cc

```diff
@@ -7748,6 +7748,18 @@ static bool fil_keyring_skip_encryption(const page_id_t &page_id) {
 @param[in]	space		table space */
 void fil_io_set_encryption(IORequest &req_type, const page_id_t &page_id,
                            fil_space_t *space) {
+  // TODO: now that dblwr doesn't exist in sys , should we encrypt all pages?
```

Comment:
Original patch also used to have `req_type.is_encryption_disabled()` check here.
Any reason why it is not needed any more?

Reply:
I did a bad job not commenting on why I introduced this flag in IOrequest.

When system tablespace is encrypted, all pages in system tablespace dblwr would be written encrypted too. Page0 of every tablespace should be unencrypted. When such a page is written to dblwr, it is written at doublewrite buffer offset. the page_id used is different. It is (0, offset_in_dblwr).  The check in fil_io_set_encryption() will fail to skip encryption for these i pages in system dblwr.

Hence, for these special pages which are always supposed to be un-encrypted, we force them to be un-encrypted in system tablesapce dblwr.

SInce the dblwr in system tablespace is gone, this change is no longer required.